openedx/core/djangoapps/credit/serializers.py

```diff
@@ -2,12 +2,33 @@
 
 from rest_framework import serializers
 
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys import InvalidKeyError
 from openedx.core.djangoapps.credit.models import CreditCourse
 
 
+class CourseKeyField(serializers.Field):
+    """
+    Serializer field for a model CourseKey field.
+    """
+
+    def to_representation(self, data):
+        """Convert a course key to unicode. """
+        return unicode(data)
+
+    def to_internal_value(self, data):
+        """Convert unicode to a course key. """
+        try:
+            return CourseKey.from_string(data)
+        except InvalidKeyError as ex:
+            raise serializers.ValidationError("Invalid course key: {msg}".format(msg=ex.msg))
```

Comment:
This message will get seen by a user if they try to put in an invalid course key, right? If so, then it needs to be internationalized (wrapped in `ugettext` calls).